File: servers/assessment/assessmentTemplates/assessmentTemplateDTO/assessmentTemplate.go
package assessmentTemplateDTO

import (
	"time"

	"github.com/google/uuid"
	db "github.com/ls1intum/prompt2/servers/assessment/db/sqlc"
)

type AssessmentTemplate struct {
	ID          uuid.UUID `json:"id"`
	Name        string    `json:"name"`
	Description string    `json:"description"`
	CreatedAt   time.Time `json:"createdAt"`
	UpdatedAt   time.Time `json:"updatedAt"`
}

func MapDBAssessmentTemplateToDTOAssessmentTemplate(dbTemplate db.AssessmentTemplate) AssessmentTemplate {
	return AssessmentTemplate{
		ID:          dbTemplate.ID,
		Name:        dbTemplate.Name,
		Description: dbTemplate.Description.String,
		CreatedAt:   dbTemplate.CreatedAt.Time,
		UpdatedAt:   dbTemplate.UpdatedAt.Time,
	}
}

File: clients/assessment_component/src/assessment/interfaces/assessmentTemplate.ts
export interface AssessmentTemplate {
  id: string
  name: string
  description: string
  createdAt?: Date
  updatedAt?: Date
}

export interface CreateAssessmentTemplateRequest {
  name: string
  description: string
}

export interface UpdateAssessmentTemplateRequest {
  name: string
  description: string
}

export interface CreateOrUpdateAssessmentTemplateCoursePhaseRequest {
  assessmentTemplateID: string
  coursePhaseID: string
}
